crates/executor/src/executor.rs

```diff
@@ -17,7 +18,157 @@ use revm::{
+        &mut self,
+        block: &Block,
+        signers: Option<Vec<Address>>,
+    ) -> Result<ExecutionResult, Error> {
+        let Block { header, body, ommers } = block;
+        let signers = self.recover_signers(body, signers)?;
+
+        self.init_block_env(header);
+
+        let mut cumulative_gas_used = 0;
+        // output of verification
+        let mut changesets = Vec::with_capacity(body.len());
+
+        for (transaction, signer) in body.iter().zip(signers.into_iter()) {
+            // The sum of the transaction’s gas limit, Tg, and the gas utilised in this block prior,
+            // must be no greater than the block’s gasLimit.
+            let block_available_gas = header.gas_limit - cumulative_gas_used;
+            if transaction.gas_limit() > block_available_gas {
+                return Err(Error::TransactionGasLimitMoreThenAvailableBlockGas {
+                    transaction_gas_limit: transaction.gas_limit(),
+                    block_available_gas,
+                })
+            }
+
+            // Fill revm structure.
+            revm_wrap::fill_tx_env(&mut self.evm.env.tx, transaction, signer);
+
+            // Execute transaction.
+            let out = self.evm.transact();
+
+            // Useful for debugging
+            // let out = evm.inspect(revm::inspectors::CustomPrintTracer::default());
+            // tracing::trace!(target:"evm","Executing transaction {:?}, \n:{out:?}: {:?}
+            // \nENV:{:?}",transaction.hash(),transaction,evm.env);
+
+            let (revm::ExecutionResult { exit_reason, gas_used, logs, .. }, state) = out;
+
+            // Fatal internal error.
+            if exit_reason == revm::Return::FatalExternalError {
+                return Err(Error::ExecutionFatalError)
+            }
+
+            // Success flag was added in `EIP-658: Embedding transaction status code in receipts`.
+            // TODO for verification (exit_reason): some error should return EVM error as the block
+            // with that transaction can have consensus error that would make block
+            // invalid.
+            let is_success = match exit_reason {
+                revm::return_ok!() => true,
+                revm::return_revert!() => false,
+                _ => false,
+                //e => return Err(Error::EVMError { error_code: e as u32 }),
+            };
+
+            // Add spend gas.
+            cumulative_gas_used += gas_used;
+
+            // Transform logs to reth format.
+            let logs: Vec<Log> = logs
+                .into_iter()
+                .map(|l| Log {
+                    address: H160(l.address.0),
+                    topics: l.topics.into_iter().map(|h| H256(h.0)).collect(),
+                    data: l.data.into(),
+                })
+                .collect();
+
+            // commit state
+            let (changeset, new_bytecodes) =
+                commit_changes(self.evm.db().expect("Db to not be moved."), state);
```

Comment:
Unsure if we want the executor itself to commit the state since the idea here is that it can be re-used to simulate blocks in e.g. the RPC. Not sure what the best way of handling this is - a committing and non-committing variant of the executor? Or perhaps a `commit(changeset)` function on the trait itself

Reply:
follow it up w/ 553c0bb846dd2f810e9e99bba975f88ab39e7137

I'll try to move more functions to `Executor`'s method.

Reply:
`commit_changes` here commits the transaction data to substate cache of database (revm in memory db) and creates the changeset out of it.
For block execution, that this is this function, it is necessary.

Reply:
@rakita Sorry, I couldn't catch your point. Do you mean we need to keep `commit_changes()` as a standalone function? If it is, I'll revert my change.

Reply:
You can leave it as is, it is needed to update the state between transactions.